istioctl/pkg/rbac/upgrade_test.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rbac
+
+import (
+	"io/ioutil"
+	"reflect"
+	"testing"
+)
+
+const (
+	testFailedWithError   = "test failed with error %v"
+	testFailedExpectedGot = "test failed. Expected\n%sGot\n%s"
+)
+
+type testCases struct {
+	input                string
+	workloadLabelMapping map[string]ServiceToWorkloadLabels
+	expected             string
+}
+
+func TestUpgradeLocalFile(t *testing.T) {
+	cases := []testCases{
+		{
+			input: "./testdata/rbac-policies.yaml",
+			// Data from the BookExample. productpage.svc.cluster.local is the service with pod label
+			// app: productpage.
+			workloadLabelMapping: map[string]ServiceToWorkloadLabels{
+				"service-viewer": {
+					"productpage": map[string]string{
+						"app": "productpage",
+					},
+				},
+			},
+			expected: "./testdata/rbac-policies-v2.golden.yaml",
+		},
+	}
```

Comment:
Can you add a test case with `paths` and `methods` in the input?

Reply:
Okay will do soon.

Reply:
Done.